CONTRIBUTING.md

```diff
@@ -0,0 +1,104 @@
+Contributing to Bitcoin Core
+============================
+
+The Bitcoin Core project operates an open contributor model where anyone is welcome to contribute towards development in the form of peer review, testing and patches. This document explains the practical process and guidelines for contributing.
+
+Firstly in terms of structure, there is no particular concept of “Core developers” in the sense of privileged people. Open source often naturally revolves around meritocracy where longer term contributors gain more trust from the developer community. However, some hierarchy is necessary for practical purposes. As such there are repositories “maintainers” who are responsible for merging pull requests as well as a “lead maintainer” who is responsible for overall merging, moderation and appointment of maintainers and the release cycle.
+
+
+Contributor Workflow
+--------------------
+
+The codebase is maintained using the “contributor workflow” where everyone without exception contributes patch proposals using “pull requests”. This facilitates social contribution, easy testing and peer review.
+
+  - Fork repository
+  - Create topic branch
+  - Commit patches 
+
+The project coding conventions in [doc/developer-notes.md](doc/developer-notes.md) must be adhered to.
+
+In general [commits should be atomic](https://en.wikipedia.org/wiki/Atomic_commit#Atomic_commit_convention) and diffs should be easy to read. For this reason do not mix any formatting fixes or code moves with actual code changes.
+
+Commit messages should be verbose by default consisting of a short subject line (50 chars max), a blank line and detailed explanatory text as separate paragraph(s); unless the title alone is self-explanatory (like "Corrected typo in main.cpp") then a single title line is sufficient. Commit messages should be helpful to people reading your code in the future, so explain the reasoning for your decisions. Further explanation [here](http://chris.beams.io/posts/git-commit/).
+
+If a particular commit references another issue, please add the reference, for example "refs #1234", or "fixes #4321". Using "fixes or closes" keywords will cause the corresponding issue to be closed when the pull request is merged.
+
+Please refer to the [Git manual](https://git-scm.com/doc) for more information about Git.
+
+  - Push changes to your fork
+  - Create pull request
+
+The title of the pull request should be prefixed by the component or area that the pull request affects. Examples:
+
+    [P2P] Add ZMQ message support
+    [Consensus] Add new opcode for BIP-XXXX OP_CHECKAWESOMESIG
+    [Qt Wallet] Refactor send code
```

Comment:
Prefer colon format (and seems to be more common):

    p2p: Add ZMQ message support
    consensus: Add new opcode for BIP-XXXX
    qt/wallet: Refactor send code

(also note ZMQ really has nothing to do with p2p..)